src/managerHistory.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * @typedef ManagerHistory
```

Comment:
Hmmm maybe we can find a better name for this class, but for now I think `HistoryManager` should be the correct one, what do you think?